core/fake/mutation_storage.go

```diff
@@ -16,50 +16,50 @@ package fake
 
 import (
 	"context"
+	"fmt"
 
 	pb "github.com/google/keytransparency/core/api/v1/keytransparency_proto"
-	"github.com/google/keytransparency/core/mutator"
 )
 
 // MutationStorage implements mutator.Mutation
 type MutationStorage struct {
-	mtns map[string][]*pb.EntryUpdate
+	// mtns[DomainID][Epoch]
+	mtns map[string]map[int64][]*pb.Entry
 }
 
 // NewMutationStorage returns a fake mutator.Mutation
 func NewMutationStorage() *MutationStorage {
 	return &MutationStorage{
-		mtns: make(map[string][]*pb.EntryUpdate),
+		mtns: make(map[string]map[int64][]*pb.Entry),
 	}
 }
 
 // ReadPage paginates through the list of mutations
-func (m *MutationStorage) ReadPage(_ context.Context, domainID string, start, end int64, pageSize int32) (int64, []*pb.Entry, error) {
-	if start > int64(len(m.mtns[domainID])) {
-		panic("start > len(m.mtns[domainID])")
+func (m *MutationStorage) ReadPage(_ context.Context, domainID string, revision, start int64, pageSize int32) (int64, []*pb.Entry, error) {
+	domain, ok := m.mtns[domainID]
+	if !ok {
+		return 0, nil, fmt.Errorf("DomainID %v not found", domainID)
 	}
-	// Adjust end.
-	if end-start > int64(pageSize) {
-		end = start + int64(pageSize)
+	mutationList, ok := domain[revision]
+	if !ok {
+		return 0, nil, fmt.Errorf("DomainID: %v, revision %v not found", domainID, revision)
 	}
-	if end > int64(len(m.mtns[domainID])) {
-		end = int64(len(m.mtns[domainID]))
+	if got, want := int(start), len(mutationList); got >= want {
```

Comment:
I think the 'got' / 'want' idiom obscures the intent of your code.
I wouldn't use it outside of test code personally.

Reply:
I was originally using `got` and `want` inside the if block, but now that that's no longer the case, you're right, it obscures the code's meaning.